### PR TITLE
fix(tooling): upgrading node version to 20.13

### PR DESCRIPTION
**Pull request by Kesari3008**
Upgrading the node version from 14 to 20.13.
Also updated deasync package as older version was not compatible with node 20.
Updated the circleci config as in pipeline unit test and integration test started failing due to node environment being used was for 14, not 20.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,15 +23,15 @@ commands:
...
+    executor: node-iron-executor
     steps:
       - checkout
       - restore_cache:
           keys:
             - dependency-cache-{{ checksum "package-lock.json" }}
       - run:
           name: Install Dependencies & Peer Dependencies
-          command: npx npm-install-peers
+          command: npm install
```
[the remark this answers is not in the document]
Comment (Kesari3008) [reply .circleci/config.yml:34]: Now that we have npm version 10.x being used, peer dependencies get installed with npm install itself

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,14 +2,14 @@ version: 2.1
 
 # reusable environment for all jobs
 executors:
-  node-fermium-executor:
```
[the remark this answers is not in the document]
Comment (Kesari3008) [reply .circleci/config.yml:5]: fermium is node 14 and iron is node 20. It's their codename

File: package.json [hunk 3]
```diff
@@ -58,7 +59,7 @@
     "babel-plugin-istanbul": "^6.1.1",
     "cypress": "^9.5.3",
     "dotenv": "^8.2.0",
-    "eslint": "^7.2.0",
+    "eslint": "^7.32.0",
     "eslint-config-airbnb-base": "^14.2.1",
     "eslint-plugin-cypress": "^2.12.1",
     "eslint-plugin-import": "^2.22.1",
```
Comment (Neeraj-swarnkar) [inline package.json:65]: @Kesari3008,
Should we add engine details in package.json
```
"engines": {
        "node": ">=20.13"
    }
```
and refer node version in .nvmrc

`20`

Did you verify if in local development node version is selected as 20+ while running project ?

**Review by Neeraj-swarnkar — COMMENTED** [review]
Added some comments for clarification,

File: package.json (hunk 3 above)
Comment (Kesari3008) [reply package.json:65]: @Neeraj-swarnkar Updated the engine details, .nvmrc file is already updated with the codename of node 20. I checked the node version using command node -v while running npm install and npm run build and in the circle ci pipeline is updated to use node 20.13. Is there any other way you would like me top verify if node 20+ is being used in local development.

Comment (Neeraj-swarnkar) [reply package.json:68]: @Kesari3008, Thanks for making the changes,

For local development, Same command using  `node -v `
Also if nvm present `nvm ls ` will tell which one is currently active node version,
Switch to some other node version using nvm use and do yarn install or npm install and check nvm ls

**Review by sreenara — APPROVED** [review]

**Review by Neeraj-swarnkar — APPROVED** [review]
Looks Good.

**Pull request merged**